### ISSUE TITLE
Please allow 'acl' to defaults section

### ISSUE THREAD

**Issue by jsalatiel** [haproxy/haproxy#1820]
### Your Feature Request

It would be nice to be able to have acl as a default parameter so it would be applied to any frontend/backend.

### What are you trying to do?

I have an  acl 
```
acl rfc1918 src 10.0.0.0/8 172.16.0.0/12 192.168.0.0/16
```
I have to replicate that for every new frontend and backend I create to be able to use it.
It would be great if I could just have it inside a global or default section

### Output of `haproxy -vv`

```plain
HAProxy version 2.5.8-0cbd0f6 2022/07/25 - https://haproxy.org/
```

**Comment by wuestkamp** [comment]
Is there another recommendation for having the same ACLs for different frontends without having to duplicate?

**Comment by capflam** [comment]
Since the 2.6, it is possible to define ACL in defaults sections. But only in named defaults sections. I guess we can close this FR.

**Comment by dosmanak** [comment]
Is there any documentation on this?
I have [found](http://docs.haproxy.org/2.8/configuration.html#4.2-acl)
> This directive is only available from named defaults sections, not anonymous
> ones. ACLs defined in a defaults section are not visible from other sections
> using it.

But I do not understand the concept of named and anonymous default sections.

**Comment by jsalatiel** [comment]
@capflam  I do not think that's working in > 2.6.
Simple haproxy.cfg
```
global
    log stdout local0

defaults http_defaults
  mode http
  acl rfc1918 src 10.0.0.0/8 172.16.0.0/12 192.168.0.0/16

frontend admin-status from http_defaults
    maxconn 1000
    bind *:8080
    stats enable
    stats show-legends
    stats uri /
    stats refresh 10s
    http-request deny if !rfc1918
```

Log shows:
```
2024-08-14T07:44:42.071290844-03:00 stderr F [NOTICE]   (1) : haproxy version is 2.6.18-bb135ab
2024-08-14T07:44:42.071290844-03:00 stderr F [NOTICE]   (1) : path to executable is /usr/local/sbin/haproxy
2024-08-14T07:44:42.071290844-03:00 stderr F [ALERT]    (1) : config : parsing [/usr/local/etc/haproxy/haproxy.cfg:15] : error detected while parsing an 'http-request deny' condition : no such ACL : 'rfc1918'.
2024-08-14T07:44:42.071290844-03:00 stderr F [ALERT]    (1) : config : Error(s) found in configuration file : /usr/local/etc/haproxy/haproxy.cfg
```


Same config on 2.8 also fails.
```
2024-08-14T07:46:07.565009843-03:00 stderr F [NOTICE]   (1) : haproxy version is 2.8.10-f28885f
2024-08-14T07:46:07.565009843-03:00 stderr F [NOTICE]   (1) : path to executable is /usr/local/sbin/haproxy
2024-08-14T07:46:07.565009843-03:00 stderr F [ALERT]    (1) : config : parsing [/usr/local/etc/haproxy/haproxy.cfg:15] : error detected while parsing an 'http-request deny' condition : no such ACL : 'rfc1918'.
2024-08-14T07:46:07.565009843-03:00 stderr F [ALERT]    (1) : config : Error(s) found in configuration file : /usr/local/etc/haproxy/haproxy.cfg
```

**Comment by Darlelet** [comment]
Hello @jsalatiel 

You need to actually write the rule in the default section, not just the condition, for it to be inherited by proxies referencing the
named default section:
```haproxy
global
    log stdout local0

defaults http_defaults
  mode http
  acl rfc1918 src 10.0.0.0/8 172.16.0.0/12 192.168.0.0/16
  http-request deny if !rfc1918

frontend admin-status from http_defaults
    maxconn 1000
    bind *:8080
    stats enable
    stats show-legends
    stats uri /
    stats refresh 10s
```

**Comment by Darlelet** [comment]
Logic was implemented in ee08d6cc74daef511ce2f8d7e3f766fb6c9ae385 and documented in 6e0425b71878c5aa470ecf894f9fbdd2526f770c

ACL are not actually inherited, it's the rules themselves that may be inherited
